pkg/resourcemonitor/resourcemonitor.go

```diff
@@ -92,30 +92,34 @@ type resourceMonitor struct {
 	nodeName          string
 	args              Args
 	podResCli         podresourcesapi.PodResourcesListerClient
+	k8sCli            kubernetes.Interface
 	topo              *ghw.TopologyInfo
 	coreIDToNodeIDMap map[int]int
 	nodeCapacity      perNUMAResourceCounter
 	nodeAllocatable   perNUMAResourceCounter
 }
 
-func NewResourceMonitor(podResCli podresourcesapi.PodResourcesListerClient, args Args) (*resourceMonitor, error) {
-	topo, err := ghw.Topology(ghw.WithPathOverrides(ghw.PathOverrides{
-		"/sys": args.SysfsRoot,
-	}))
-	if err != nil {
-		return nil, err
+func NewResourceMonitor(podResCli podresourcesapi.PodResourcesListerClient, args Args, options ...func(*resourceMonitor)) (*resourceMonitor, error) {
+	rm := &resourceMonitor{
+		podResCli: podResCli,
+		args:      args,
+	}
+	for _, o := range options {
+		o(rm)
 	}
-	nodeName := os.Getenv("NODE_NAME")
-	return NewResourceMonitorWithTopology(nodeName, topo, podResCli, args)
-}
 
-func NewResourceMonitorWithTopology(nodeName string, topo *ghw.TopologyInfo, podResCli podresourcesapi.PodResourcesListerClient, args Args) (*resourceMonitor, error) {
-	rm := &resourceMonitor{
-		nodeName:          nodeName,
-		podResCli:         podResCli,
-		args:              args,
-		topo:              topo,
-		coreIDToNodeIDMap: MakeCoreIDToNodeIDMap(topo),
+	if rm.topo == nil {
+		topo, err := ghw.Topology(ghw.WithPathOverrides(ghw.PathOverrides{
+			"/sys": args.SysfsRoot,
+		}))
+		if err != nil {
+			return nil, err
+		}
+		rm.topo = topo
```

Comment:
missing
```
rm.coreIDToNodeIDMap = MakeCoreIDToNodeIDMap(topo)
```
?

Reply:
I think that is the bug I'm struggling with :)